flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2326,6 +2340,52 @@ mlir::Value IntrinsicLibrary::genModulo(mlir::Type resultType,
   return builder.create<mlir::SelectOp>(loc, mustAddP, remPlusP, remainder);
 }
 
+// MVBITS
+void IntrinsicLibrary::genMvbits(llvm::ArrayRef<fir::ExtendedValue> args) {
+  // A conformant MVBITS(FROM,FROMPOS,LEN,TO,TOPOS) call satisfies:
+  //     FROMPOS >= 0
+  //     LEN >= 0
+  //     TOPOS >= 0
+  //     FROMPOS + LEN <= BIT_SIZE(FROM)
+  //     TOPOS + LEN <= BIT_SIZE(TO)
+  // MASK = -1 >> (BIT_SIZE(I) - LEN)
+  // TO = LEN == 0 ? TO : ((!(MASK << TOPOS)) & TO) |
+  //                      (((FROM >> FROMPOS) & MASK) << TOPOS)
```

Comment:
Are we checking all these conditions in the inline code? I only see 1 compare instruction ...

Would it be worth noting here in the comment that if the values are out of range, the result of MVBITS is undefined, implementation defined, or some such?

Reply:
You've got it.  For all of these bit intrinsics (probably others as well) most compiler implementations agree on results for valid input, but then generate "whatever happens" results for nonconformant input.  There is some degree of agreement across compilers for nonconformant results, but plenty of diversity in results as well.